### PR TITLE
auth checking and redirection

### PR DESCRIPTION
**Pull request by defintelyNotKabi**
https://trello.com/c/KGD0HSYn/112-auth-checking-and-redirection

### CODE REVIEW

File: Frontend/Frontend.Client/Models/WebSocketService.cs [hunk 1]
```diff
@@ -91,4 +99,4 @@ public async Task SendMessageAsync(string message)
             await _webSocket.SendAsync(new ArraySegment<byte>(bytes), WebSocketMessageType.Text, true, CancellationToken.None);
         }
     }
-}
+}
```
Comment (OmarQurashi868) [inline Frontend/Frontend.Client/Models/WebSocketService.cs:129]: new line where?

File: Frontend/Frontend.Client/Pages/LoginPage.razor [hunk 2]
```diff
@@ -77,7 +77,24 @@
...
+
+            isLoading = false;
+
+            if (WebSocketService.IsAuthenticated)
+            {
+                NavigationManager.NavigateTo("main-page");
+            }
+            else
+            {
+                errorMesages.Add("Failed to authenticate. Please check your credentials.");
```
Comment (OmarQurashi868) [inline Frontend/Frontend.Client/Pages/LoginPage.razor:82]: The server returns an error message with the REFUSE, why are we not using it here?

File: Frontend/Frontend.Client/Models/WebSocketService.cs [hunk 3]
```diff
@@ -29,35 +35,37 @@ public async Task ConnectAsync(string uri)
...
         {
-            var lines = message.Split(new[] { "\r\n", "\n" }, StringSplitOptions.None);
-
-            if (lines.Length >= 6 &&
-                lines[0].StartsWith("DO SEND") &&
-                lines[1].StartsWith("FROM ") &&
-                lines[2].StartsWith("IN general-chat") &&
-                lines[3].StartsWith("AT ") &&
-                lines[4].StartsWith("WITH"))
+            if (message.StartsWith("DO ACCEPT"))
```
Comment (ahmed-okkio) [inline Frontend/Frontend.Client/Models/WebSocketService.cs]: We send DO ACCEPT as a response for many things, you'll need to specify this DO ACCEPT behind another condition ensuring its the DO ACCEPT related to the AUTH request

**Comment by ahmed-okkio** [comment]
Not able to understand the changes we should discuss them in a call

**Comment by OmarQurashi868** [comment]
kamen-rider-wizard-washing

**Review by MidfieId — APPROVED** [review]

**Pull request merged**